### ISSUE TITLE
Cannot repeat the same r2 on Zrimec yeast expression task.

### ISSUE THREAD

**Issue by Azai-yx** [gagneurlab/SpeciesLM#4]
Hello, Great works!

However, when I tried to use representations from SpeciesLM and directly applied them to predict Zrimec task, I can not achieve the same r2 that reported in your paper.

`
import numpy as np
import pandas as pd
import h5py
import matplotlib.pyplot as plt
from sklearn.preprocessing import StandardScaler
from sklearn.linear_model import LinearRegression, RidgeCV
from sklearn.pipeline import make_pipeline
from sklearn.model_selection import train_test_split
from scipy.stats import pearsonr, spearmanr

def r2_ss(y_true, y_pred):
    y_true = np.asarray(y_true)
    y_pred = np.asarray(y_pred)
    ss_res = np.sum((y_true - y_pred) ** 2)
    ss_tot = np.sum((y_true - y_true.mean()) ** 2)
    return 1 - ss_res / ss_tot

X_embed = embeddings     

print(X_embed.shape)

y = dataset["TPM_box_cox_transformed"].to_numpy()     
print(y.shape)

X_train, X_test, y_train, y_test = train_test_split(
    X_embed, y, test_size=0.20, random_state=42, shuffle=True
)

pipe = make_pipeline(
    StandardScaler(),
    RidgeCV(alphas=[10, 100, 1000], cv=5, scoring="neg_mean_squared_error")
)
pipe.fit(X_train, y_train)                              
best_alpha = pipe.named_steps["ridgecv"].alpha_         

y_pred = pipe.predict(X_test)
pcc,  _ = pearsonr(y_test, y_pred)
spcc, _ = spearmanr(y_test, y_pred)
r2 = r2_ss(y_test, y_pred)

print(f" alpha: {best_alpha}")
print(f"PCC  = {pcc:.4f}")
print(f"SPCC = {spcc:.4f}")
print(f"r2 = {r2:.4f}")

plt.figure(figsize=(6, 6))
plt.scatter(y_test, y_pred, s=12, alpha=0.5)
plt.plot([y_test.min(), y_test.max()], [y_test.min(), y_test.max()], "--")
plt.xlabel("True boxcox_TPM (test)")
plt.ylabel("Predicted")
plt.title(f"RidgeCV (alpha={best_alpha})\nPCC={pcc:.2f}  SPCC={spcc:.2f}")
plt.tight_layout()
plt.savefig("ridgecv_test_scatter.png", dpi=300)
plt.show()
`

And this is my result:
`alpha: 1000
PCC  = 0.4809
SPCC = 0.3952
r2 = 0.2196`

![Image](https://github.com/user-attachments/assets/84846548-2637-44c3-8f31-715608711975)

I was using Zrimec dataset however randomly splited, with seed 42 and split size train:test 0.8:0.2.
And the representation was using 1000bp 5' upstream sequences as input, with parameters below:
`kmer_size = 6 # size of kmers, always 6
proxy_species = "kazachstania_africana" # species token to use
pred_batch_size = 128*3 # batch size for rolling masking
target_layer = (8,) # what hidden layers to use for embedding`

Could you help me find out what steps were missing?

**Comment by Karollus** [comment]
Hi,

The tokenizer is unfortunately a bit sensitive to the exact species names used in the ensemble data. Thus it should be

proxy_species = "kazachstania_africana_cbs_2517_gca_000304475"

If not this exact name is used, it will get tokenized as the unknown token, which likely explains the difference in performance. You can check all the exact species names the tokenizer knows using tokenizer.get_added_vocab

The following notebook will allow to reproduce the Zrimec result from the paper: https://drive.google.com/file/d/11521ZehzV7HovxPT45OYntV0nQTEAnS8/view?usp=sharing

**Comment by Azai-yx** [comment]
Thanks! This has been very useful!
